cdm/core/src/test/java/ucar/nc2/util/cache/TestS3NetcdfFileCache.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 1998-2022 University Corporation for Atmospheric Research/Unidata
+ * See LICENSE for license information.
+ */
+package ucar.nc2.util.cache;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.BeforeClass;
+import org.junit.AfterClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.nc2.dataset.DatasetUrl;
+import ucar.nc2.dataset.NetcdfDatasets;
+import ucar.nc2.util.CancelTask;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import ucar.unidata.util.test.category.NeedsUcarNetwork;
+import ucar.unidata.util.test.category.NeedsExternalResource;
+
+/** Test FileCache. */
+@Category(NeedsExternalResource.class)
```

Comment:
This may be used in some of our tests but should not be necessary in order to perform S3 tests. We have a `thredds-test-data` public test bucket that should work for this. For instance, if you just need a single s3 file you could use the key `"cdms3:thredds-test-data?testData.nc"`

Reply:
ah, I'll use that, thanks!